### DISCUSSION TITLE
Build dependencies

### DISCUSSION THREAD

**Discussion by GeorgJung** [wfx/teg#51] [Q&A] [answered]
Hi all,
I am trying to compile T.E.G. from the sources. I keep running into missing libraries and tools during autoconf. Can somebody make a list of packages which need to be installed to compile T.E.G.?

**Comment by wfx** [comment, marked as answer]
Hello, take a look in the docker directory. There you will find dependencies for Debian and SuSE.

**Reply by GeorgJung** [reply]
Thanks @wfx! That was helpful, but hard to find.

If I may shoot the next question: After installing the dependencies as seen in the docker directory I ran `build` in the root directory of the teg-master branch. Compiled smoothly and tested successfully. However, it is unclear to me how to install from there. Or was running `build` the wrong approach in first place?

**Reply by wfx** [reply]
Depending on whether you want to install it like any other application, then in the teg source directory a .configure && make && make install is sufficient. The way to do it via build is described in the HACKING.gst.

**Reply by GeorgJung** [reply]
Thanks again @wfx. This was really helpful. The contents of that file should be in the Readme!

With your help I was now able to compile T.E.G. completely. I will however have to file a bug report soon, just like an older version T.E.G. crashes when I successfully attack with one army...

**Reply by lisandromc** [reply]
Hi! To add to the discussion: I didn't spot the `docker` directory myself, so I went through the whole process of installing dependencies manually, in my case for Linux Mint 22. I've noticed a few differences from what's in `docker/debian`:

```
apt install libtool autopoint gettext libglib2.0-dev libxml2-dev libgoocanvasmm-2.0-dev xmlto libgtest-dev libgmock-dev
```

So in my case it seems I was also missing:

- libtool
- autopoint
- libglib2.0-dev
- libgtest-dev

On the subject of the crash: @GeorgJung This also happens to me with v0.13 and v0.12. I filed an issue #53 . Please consider adding more input to that.

**Reply by wfx** [reply]
Issue #53 is closed.
We have a new documentation that still needs some revision. If anyone wants to help, we'd be very happy to help.

Thx a lot.
